### PR TITLE
feat(oas): add support the file type defined in Swagger 2.0

### LINKED ISSUE #59: Add support the file type defined in Swagger 2.0
**Description**

Currently, `@har-sdk/oas` only supports a limited set of file types. To provide better support for Swagger 2.0, we would like to add the ability to handle the `file` types defined in Swagger 2.0. This would allow our users to more easily work with their Swagger 2.0 defined APIs.

**Example**
```yaml
swagger: '2.0'
info:
  title: Example API
  version: '1.0.0'
consumes:
  - application/json
  - multipart/form-data
produces:
  - application/json
paths:
  /example:
    post:
      operationId: addExample
      consumes:
        - multipart/form-data
      parameters:
        - name: file
          in: formData
          type: file
          required: true
      responses:
        201:
          description: Example added successfully
```

### PR DESCRIPTION
**Pull request by derevnjuk**
closes #59

### CODE REVIEW

File: packages/oas/src/converter/parts/postdata/BodyConverter.ts [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+        const headers = [
+          `Content-Disposition: form-data; name="${key}"${
+            filenameRequired ? `; filename="${key}"` : ''
+          }`,
+          ...(contentType !== 'text/plain'
+            ? [`Content-Type: ${contentType}`]
+            : []),
+          ...(this.BASE64_PATTERN.test(content) &&
+          contentType === 'application/octet-stream'
```
Comment (pmstss) [inline packages/oas/src/converter/parts/postdata/BodyConverter.ts:107]: it could lead to unwanted force of base64, e. g. providing example value `aaaa` in existing `params-form-data.swagger.yaml` results in multipart formdata entry like `Content-Type: application/octet-stream\r\nContent-Transfer-Encoding: base64\r\n\r\naaaa`

File: packages/oas/tests/fixtures/params-encoding.oas.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+                profileImage: {}
+            encoding:
+              historyMetadata:
+                contentType: application/xml; charset=utf-8
+              profileImage:
+                contentType: image/png, image/jpeg
+                headers:
+                  X-Rate-Limit-Limit:
+                    schema:
+                      type: integer
```
Comment (pmstss) [inline packages/oas/tests/fixtures/params-encoding.oas.yaml:35]: is it going to be handled? currently absent in `params-encoding.oas.result.json`

File: packages/oas/src/converter/parts/postdata/BodyConverter.ts [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+
+    switch (mime) {
+      case 'application/json':
+        return this.encodeJson(value);
+      case 'application/x-www-form-urlencoded':
+        return this.encodeFormUrlencoded(value);
+      case 'application/xml':
+        return this.encodeXml(value);
+      case 'multipart/form-data':
+      case 'multipart/mixin':
```
Comment (pmstss) [inline packages/oas/src/converter/parts/postdata/BodyConverter.ts:71]:
```suggestion
      case 'multipart/mixed':
```
also not sure that result of `encodeMultipartFormData()` is valid for "mixed" type as e. g. we have hardcoded `Content-Disposition: form-data`. imo test cases should be introduced if we are going to support it

**Review by pmstss — CHANGES_REQUESTED** [review]

File: packages/oas/src/converter/parts/postdata/BodyConverter.ts (hunk 1 above)
Comment (derevnjuk) [reply packages/oas/src/converter/parts/postdata/BodyConverter.ts:107]: This logic is inherited directly from the original [PostDataConverter](https://github.com/NeuraLegion/har-sdk/blob/99b5c4cdaf6ef07d292f912bd835dc2a5764676f/packages/oas/src/converter/parts/postdata/PostDataConverter.ts). I am not going to modify this logic at the moment because of this edge case.

File: packages/oas/tests/fixtures/params-encoding.oas.yaml (hunk 2 above)
Comment (derevnjuk) [reply packages/oas/tests/fixtures/params-encoding.oas.yaml:35]: It is out of scope this particular ticket and PR

File: packages/oas/src/converter/parts/postdata/BodyConverter.ts (hunk 3 above)
Comment (derevnjuk) [reply packages/oas/src/converter/parts/postdata/BodyConverter.ts:71]: ditto https://github.com/NeuraLegion/har-sdk/pull/176#discussion_r1109383390

File: packages/oas/tests/fixtures/params-encoding.oas.yaml (hunk 2 above)
Comment (pmstss) [reply packages/oas/tests/fixtures/params-encoding.oas.yaml:35]: should it be removed from fixture then?

Comment (derevnjuk) [reply packages/oas/tests/fixtures/params-encoding.oas.yaml:35]: I would prefer to keep it. But, nevertheless, we can remove it without any issues.

Comment (pmstss) [reply packages/oas/tests/fixtures/params-encoding.oas.yaml:35]: Created issue for this https://github.com/NeuraLegion/har-sdk/issues/180. 
Prefer to remove it from current fixture to avoid impression that those headers are already handled

Comment (derevnjuk) [reply packages/oas/tests/fixtures/params-encoding.oas.yaml]: This is your subjective impression, but we're not removing the XML models from fixtures only because they are not supported. The input data does not give any confidence about the coverage in separation from the result. Especially for that, we have opened an issue to highlight this gap.

File: packages/oas/src/converter/parts/postdata/BodyConverter.ts (hunk 1 above)
Comment (pmstss) [reply packages/oas/src/converter/parts/postdata/BodyConverter.ts:107]: Extracted to separate issue: https://github.com/NeuraLegion/har-sdk/issues/181

File: packages/oas/src/converter/parts/postdata/BodyConverter.ts (hunk 3 above)
Comment (pmstss) [reply packages/oas/src/converter/parts/postdata/BodyConverter.ts]: opened: https://github.com/NeuraLegion/har-sdk/issues/182

File: packages/oas/src/converter/parts/postdata/BodyConverter.ts [hunk 4]
```diff
@@ -0,0 +1,188 @@
+import { Sampler } from '../Sampler';
+import { SubConverter } from '../../SubConverter';
+import { OpenAPI, OpenAPIV3, PostData } from '@har-sdk/core';
+import { toXML, XmlElement } from 'jstoxml';
+import { stringify } from 'qs';
+
+export abstract class BodyConverter<T extends OpenAPI.Document>
+  implements SubConverter<PostData | null>
+{
+  private readonly JPG_IMAGE = '/9j/7g=='; // 0xff, 0xd8, 0xff, 0xee
```
Comment (pmstss) [inline packages/oas/src/converter/parts/postdata/BodyConverter.ts:10]:
```suggestion
  private readonly JPG_IMAGE = '/9j/4A=='; // 0xff, 0xd8, 0xff, 0xe0
```
as "[JFIF APP0 marker](https://en.wikipedia.org/wiki/JPEG_File_Interchange_Format#File_format_structure) (FF E0) segment immediately follows the SOI marker (FF D8)"

**Review by pmstss — APPROVED** [review]

File: packages/oas/src/converter/parts/postdata/BodyConverter.ts (hunk 4 above)
Comment (derevnjuk) [reply packages/oas/src/converter/parts/postdata/BodyConverter.ts:10]: @pmstss this corresponds to APP1 for Exif.

Comment (pmstss) [reply packages/oas/src/converter/parts/postdata/BodyConverter.ts:10]: @derevnjuk APP1 segment marker is 0xFF 0xE1 (which results in JPG_IMAGE = `/9j/4Q==`)

Comment (derevnjuk) [reply packages/oas/src/converter/parts/postdata/BodyConverter.ts:10]: @pmstss you are right, it's my fault. yet, as I mentioned above it is still a valid jpeg. The marker ending can be between 0xe0 and 0xef.
